common/src/main/java/com/wynntils/screens/itemsharing/SavedItemsScreen.java

```diff
@@ -0,0 +1,607 @@
+        // Enter can also be used to submit name for new category title
+        if ((addingCategory || editingCategory) && keyCode == GLFW.GLFW_KEY_ENTER) {
+            addCategory();
+
+            if (addingCategory) {
+                addingCategory = false;
+            } else if (editingCategory) {
+                editingCategory = false;
+            }
+            return false;
+        }
+
+        return super.keyPressed(keyCode, scanCode, modifiers);
+    }
+
+    public void scrollCategories(int scrollDirection) {
+        List<String> categories = new ArrayList<>(Services.ItemVault.categories.get());
+
+        int currentIndex = categories.indexOf(currentCategory);
+        int newIndex = currentIndex + scrollDirection;
+
+        // If scrolling past one end of the set, go to the other end
+        if (newIndex >= categories.size()) {
+            currentCategory = categories.get(0);
+        } else if (newIndex < 0) {
+            currentCategory = categories.get(categories.size() - 1);
+        } else {
+            currentCategory = categories.get(newIndex);
+        }
+
+        itemScrollOffset = 0;
+        populateItems();
+    }
+
+    private void renderScrollButton(PoseStack poseStack) {
+        float renderX = this.leftPos + 153;
+        float renderY =
+                this.topPos + 18 + MathUtils.map(itemScrollOffset, 0, getMaxScrollOffset(), 0, SCROLL_AREA_HEIGHT);
+
+        RenderUtils.drawTexturedRect(poseStack, Texture.VAULT_SCROLL, renderX, renderY);
+    }
+
+    private void addCategoryInput() {
+        // Remove previous widget if it exists
+        if (categoryInput != null) {
+            this.removeWidget(categoryInput);
+        }
+
+        categoryInput = new TextInputBoxWidget(
+                this.leftPos + 36, this.topPos + 5, 99, 16, null, (TextboxScreen) McUtils.mc().screen, categoryInput);
+
+        // Get the current category name and add if we are editing
+        if (editingCategory) {
+            categoryInput.setTextBoxInput(currentCategory);
+        }
+
+        this.addRenderableWidget(categoryInput);
+    }
+
+    private void addCategory() {
+        String newCategory = categoryInput.getTextBoxInput();
+
+        // Remove the input widget
+        this.removeWidget(categoryInput);
+        categoryInput = null;
+
+        if (newCategory.isEmpty()) return;
+
+        if (addingCategory) {
+            // Save new category
+            Services.ItemVault.categories.get().add(newCategory);
+            Services.ItemVault.categories.touched();
+
+            if (!selectedItems.isEmpty()) {
+                currentCategory = newCategory;
+                moveSelectedItems();
+            }
+        } else if (editingCategory) {
+            // Add renamed category and remove previous name
+            Services.ItemVault.categories.get().add(newCategory);
+            Services.ItemVault.categories.get().remove(currentCategory);
```

Comment:
This does not preserve the category order, maybe we should do that?

Reply:
Currently they're just ordered A-z. Do we want to be able to customise the order, could make right clicking the navigaton buttons shift their position or something

Reply:
Hm, okay, that should work.